src/Bicep.Core/Syntax/IfExpressionSyntax.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class IfExpressionSyntax : ExpressionSyntax
+    {
+        public IfExpressionSyntax(Token keyword, SyntaxBase conditionExpression, SyntaxBase consequenceExpression)
+        {
+            this.Keyword = keyword;

```

Comment:
>this [](start = 12, length = 4)

Can you add asserts to guard token type and token text? Can probably also assert on condition and consequence being derived from ExpressionSyntax. There are some protected methods in the base class for this that you can probably reuse. 

(These objects can be constructed manually in syntax rewriters and we will likely expose that API publically at some point, so it'll be notice to block some user mistakes.)

Reply:
Done.